### ISSUE TITLE
Hide headers with no content

### ISSUE THREAD

**Issue by julesvirallinen** [oleeskild/obsidian-digital-garden#455]
If a folder has all items as hidden, the folder will appear in the tree as empty 

<img width="218" alt="CleanShot 2023-10-02 at 16 33 35@2x" src="https://github.com/oleeskild/obsidian-digital-garden/assets/4622905/0dcf82d0-a966-4575-b7cc-46dc24a0d63a">

**Comment by dayne** [comment]
First things first: The code/repo relevant to this is actually over at https://github.com/oleeskild/digitalgarden   
Pretty sure that is the location for the relevant code and working through if this is a bug or expected behavior.  Recommend considering closing this issue here and re-opening it there. 

However, before you do that,  make sure you've updated your website template following these steps:
https://dg-docs.ole.dev/getting-started/06-updating-the-template/

However, I can confirm this behavior is still happening in latest version of the template, although I'm not sure if this is a bug. 

My vault that I have configured with a digital garden has many folders with items not included in the garden and I do not get any empty folders in the garden.  

 just played around setting `dg-publish: true`, publishing, then back to `dg-publish: false`, publishing the delete, in a deep folder that previously had not shown up on the garden website to see if that would trigger this effect.  It did not.

Then I took one of those deep folders added a published note with `dg-publish: true` and `dg-hide: true` set on just one file.  This created the effect you are talking about.

My personal reaction to this is to create some sort of index note that you don't hide that acts as a public listing in the file tree.  There is published content available just not showing up in the side bar.  Seems reasonable if you have published notes in the structure that the file tree stays consistent. 

However, if more logic is wanted to avoid this behavior then there are two places to start looking:
- [`digitalgarden/_includes/components/filetree.njk`](https://github.com/oleeskild/digitalgarden/blob/479ab3c545761994c15953bad5e0658c5ca950a5/src/site/_includes/components/filetree.njk#L52) around line 52.  That is one location you might be able to to determine if the folder is empty before passing it on.   
- And/or the building of the filetree object being used there which happens in: https://github.com/oleeskild/digitalgarden/blob/479ab3c545761994c15953bad5e0658c5ca950a5/src/helpers/filetreeUtils.js

Recommend closing up this issue and re-opening it over in the repo with relevant code:  https://github.com/oleeskild/digitalgarden

**Comment by julesvirallinen** [comment]
Yeah, I think the best would be switching to monorepo and moving all issues here, as users generally open all issues here anyway (and I think most issues for the template also are affected by the plugin side here). 

If we don't feel this is an issue we can close this though. (if it's expected behaviour)
